Index: src/game/g_active.c
===================================================================
--- src/game/g_active.c	(revision 17)
+++ src/game/g_active.c	(working copy)
@@ -1634,6 +1634,11 @@
     ent->client->pers.cmd.buttons |= BUTTON_GESTURE;
   }
 
+  if( ent->client->pers.muted )
+  {
+    ent->client->pers.cmd.buttons &= ~BUTTON_GESTURE;
+  }
+
   pm.ps = &client->ps;
   pm.pmext = &client->pmext;
   pm.cmd = *ucmd;
